PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java

```diff
@@ -195,31 +208,50 @@ private boolean isBasicAuthSelected() {
+        switch (getSparkClusterType()) {
+            case HDINSIGHT_CLUSTER:
+                if (StringUtils.isBlank(clusterNameOrUrlField.getText())) {
+                    errorMessage = "Cluster name can't be empty";
+                } else {
+                    String clusterName = ctrlProvider.getClusterName(clusterNameOrUrlField.getText());
+                    if (clusterName == null) {
+                        errorMessage = "Cluster URL is not a valid URL";
+                    } else if (ctrlProvider.doesClusterNameExistInLinkedHDInsightClusters(clusterName)) {
+                        errorMessage = "Cluster already exists in linked clusters";
+                    }
                 }
-            }
-        } else {
-            if (StringUtils.isBlank(livyEndpointField.getText()) ||
-                    StringUtils.isBlank(livyClusterNameField.getText())) {
-                errorMessage = "Livy Endpoint and cluster name can't be empty";
-            } else if (!ctrlProvider.isURLValid(livyEndpointField.getText())) {
-                errorMessage = "Livy Endpoint is not a valid URL";
-            } else if (ctrlProvider.doesClusterLivyEndpointExistInAllClusters(livyEndpointField.getText())) {
-                errorMessage = "The same name Livy Endpoint already exists in clusters";
-            } else if (ctrlProvider.doesClusterNameExistInAllClusters(livyClusterNameField.getText())) {
-                errorMessage = "Cluster Name already exists in clusters";
-            } else if (!StringUtils.isEmpty(yarnEndpointField.getText()) &&
-                    !ctrlProvider.isURLValid(yarnEndpointField.getText())) {
-                errorMessage = "Yarn Endpoint is not a valid URL";
-            }
+                break;
+            case LIVY_LINK_CLUSTER:
+                if (StringUtils.isBlank(livyEndpointField.getText()) ||
+                        StringUtils.isBlank(livyClusterNameField.getText())) {
+                    errorMessage = "Livy Endpoint and cluster name can't be empty";
+                } else if (!ctrlProvider.isURLValid(livyEndpointField.getText())) {
+                    errorMessage = "Livy Endpoint is not a valid URL";
+                } else if (ctrlProvider.doesClusterLivyEndpointExistInAllHDInsightClusters(livyEndpointField.getText())) {
+                    errorMessage = "The same name Livy Endpoint already exists in clusters";
+                } else if (ctrlProvider.doesClusterNameExistInAllHDInsightClusters(livyClusterNameField.getText())) {
+                    errorMessage = "Cluster Name already exists in clusters";
+                } else if (!StringUtils.isEmpty(yarnEndpointField.getText()) &&
+                        !ctrlProvider.isURLValid(yarnEndpointField.getText())) {
+                    errorMessage = "Yarn Endpoint is not a valid URL";
+                }
+                break;
+            case SQL_BIG_DATA_CLUSTER:
+                if (StringUtils.isBlank(livyEndpointField.getText()) ||
```

Comment:
Are these checking the same as `LIVY_LINK_CLUSTER`?

Reply:
Not exactly the same. Cluster name duplicate validation and livy endpoint duplicate validation is different for HDI livy link cluster and SQL big data cluster.

---
In reply to: [237105022](https://github.com/Microsoft/azure-tools-for-java/pull/2235#discussion_r237105022) [](ancestors = 237105022)